### PR TITLE
Kuz 692 redis cluster

### PR DESCRIPTION
**Pull request by ballinette**
Add support to Redis Cluster (cf. http://redis.io/topics/cluster-spec)
* merge all volatile cache DB into one single "internalCache" (+ add a key prefix for each usage)
* split settings of "internalCache" and "memoryStorage" to let use separate redis instance for each one (to keep descendent compatibility, it still uses the same Redis by default, with DB "0" for "internalCache", "5" for "memoryStorage" )
* add .kuzzlerc sample settings 
  * for Master/Slave with Sentinels
  * for Cluster mode

### CODE REVIEW

File: .kuzzlerc.sample [hunk 1]
```diff
@@ -204,38 +204,71 @@
...
+    // 1. using a single Redis database:
+    //   node:
+    //     * host: The host on which Redis can be reached.
+    //       Can take an IP address, an URI or a hostname
+    //     * port: The port on which Redis is running
+    //   database: (optional) the ID of the redis database (default: 0)
+    "internalCache": {
+      "backend": "redis",
+      "node": {
+        "host": "redis",
```
Comment (scottinet) [inline .kuzzlerc.sample]: Isn't this configuration re-introducing the problem detailed in #400 (and solved in #406)?

File: default.config.js [hunk 2]
```diff
@@ -125,24 +125,16 @@ module.exports = {
...
-      databases: [
-        'notificationCache',
-        'statsCache',
-        'securityCache',
-        'internalCache',
-        'tokenCache',
-        'memoryStorage'
-      ],
+      node: {
+        host: 'redis',
```
Comment (scottinet) [inline default.config.js]: Same as my comment above

File: default.config.js [hunk 3]
```diff
@@ -125,24 +125,16 @@ module.exports = {
...
+        port: 6379
+      }
+    },
+    memoryStorage: {
+      backend: 'redis',
```
Comment (scottinet) [inline default.config.js]: Ditto

File: .kuzzlerc.sample (hunk 1 above)
Comment (ballinette) [inline .kuzzlerc.sample]: you're asbolutly right ;)

Comment (ballinette) [inline .kuzzlerc.sample]: (but #406 isn't merged yet :P )

Comment (scottinet) [inline .kuzzlerc.sample]: So first merged PR wins? :stuck_out_tongue:

**Pull request merged**